lib/matplotlib/axes/_axes.py

```diff
@@ -8430,8 +8438,15 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
         elif len(widths) != N:
             raise ValueError(datashape_message.format("widths"))
 
+        # Validate side
+        if side not in ['both', 'left', 'right']:
+            raise ValueError("side should be one of 'both', 'left', or 'right'")
+
```

Comment:
```suggestion
        _api.check_in_list(["both", "left", "right"], side=side)

```

Ah... apparently I didn't click submit on this note earlier...


Essentially we just have this as a helper function that makes our error messages more consistent for these checks (which are common)

we will get:

```python
ValueError: 'foo' is not a valid value for side; supported values are 'both', 'left', 'right'
```

This has the advantage of including both the value and the parameter name, so users can more easily tell what is causing the exception.